clients/pkg/promtail/targets/kafka/target.go

```diff
@@ -46,20 +51,46 @@ func NewTarget(
 		claim:                claim,
 		session:              session,
 		client:               client,
+		relabelConfig:        relabelConfig,
 		useIncomingTimestamp: useIncomingTimestamp,
 	}
 }
 
+const (
+	defaultKafkaMessageKey  = "none"
+	labelKeyKafkaMessageKey = "__meta_kafka_message_key"
+)
+
+func (t *Target) messageKeyLabel() (useLabel bool, labelName model.LabelName) {
```

Comment:
I think we should pass the actual value to the format and not a default since relabeling can match regexes to apply change or not.

In fact all discovered labels should be there since you can compute multiple label values using `source_labels` but that sounds even trickier.

Let's start with passing the value and we'll see how we can handle the rest later. (Possibly relabel everything twice, once for the target labels, once for the message)

Reply:
Oh, you're right.

>Let's start with passing the value

Anyway, let me do that